### ISSUE TITLE
ParamValidationError: Parameter validation failed:
Invalid bucket name "": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$"

### ISSUE THREAD

**Issue by VenturaFranklin** [VenturaFranklin/thetatauCMT#187]
View details in Rollbar: [https://rollbar.com/frank.ventura/CMT/items/113/](https://rollbar.com/frank.ventura/CMT/items/113/)

```
Traceback (most recent call last):
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.6/contextlib.py", line 53, in inner
    return func(*args, **kwds)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/braces/views/_access.py", line 351, in dispatch
    request, *args, **kwargs)
  File "/home/Venturafranklin/thetatauCMT/core/views.py", line 52, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/contrib/auth/mixins.py", line 52, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/views/generic/edit.py", line 172, in post
    return super().post(request, *args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/views/generic/edit.py", line 142, in post
    return self.form_valid(form)
  File "/home/Venturafranklin/thetatauCMT/core/views.py", line 103, in form_valid
    response = super().form_valid(form)  # This saves the form
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/views/generic/edit.py", line 125, in form_valid
    self.object = form.save()
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/forms/models.py", line 458, in save
    self.instance.save()
  File "/home/Venturafranklin/thetatauCMT/thetatauCMT/submissions/models.py", line 44, in save
    super().save()
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/base.py", line 741, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/base.py", line 779, in save_base
    force_update, using, update_fields,
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/base.py", line 870, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/base.py", line 908, in _do_insert
    using=using, raw=raw)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/query.py", line 1186, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1334, in execute_sql
    for sql, params in self.as_sql():
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1278, in as_sql
    for obj in self.query.objs
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1278, in <listcomp>
    for obj in self.query.objs
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1277, in <listcomp>
    [self.prepare_value(field, self.pre_save_val(field, obj)) for field in fields]
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1228, in pre_save_val
    return field.pre_save(obj, add=True)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/fields/files.py", line 288, in pre_save
    file.save(file.name, file.file, save=False)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/db/models/fields/files.py", line 87, in save
    self.name = self.storage.save(name, content, max_length=self.field.max_length)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/django/core/files/storage.py", line 52, in save
    return self._save(name, content)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/storages/backends/s3boto3.py", line 487, in _save
    obj = self.bucket.Object(encoded_name)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/storages/backends/s3boto3.py", line 323, in bucket
    self._bucket = self._get_or_create_bucket(self.bucket_name)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/storages/backends/s3boto3.py", line 365, in _get_or_create_bucket
    bucket.meta.client.head_bucket(Bucket=name)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/client.py", line 586, in _make_api_call
    api_params, operation_model, context=request_context)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/client.py", line 619, in _convert_to_request_dict
    api_params, operation_model, context)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/client.py", line 648, in _emit_api_params
    params=api_params, model=operation_model, context=context)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File "/home/Venturafranklin/.virtualenvs/thetatauCMT/lib/python3.6/site-packages/botocore/handlers.py", line 221, in validate_bucket_name
    raise ParamValidationError(report=error_msg)
ParamValidationError: Parameter validation failed:
Invalid bucket name "": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$"
```

**Comment by VenturaFranklin** [comment]
Closed by #6475fc61a09987e4bcb31959fac284ba3f5385a9

**Comment by VenturaFranklin** [comment]
6475fc61a09987e4bcb31959fac284ba3f5385a9
